### PR TITLE
Added closest and cumulative functions

### PR DESCRIPTION
**Pull request by vetlewi**
Closing issue #144.

### CODE REVIEW

File: ompy/vector.py [hunk 1]
```diff
@@ -424,6 +424,67 @@ def to_keV(self) -> Vector:
...
+        if inplace:
+            self.E = E
+            self.values = values
+            self.std = std
+        else:
+            return Vector(values=values, E=E, std=std, units=self.units)
+
+    def cumulative(self, factor: Optional[Union[float, str]]=None,
+                   inplace=False) -> Union[Vector, None]:
+        """ Calculate the cumulative sum of the vector.
```
Comment (fzeiser) [inline ompy/vector.py:457]: Suggestion: Delete "Calculate the"  -- might make it clearer that the `inplace` option transforms the vector

File: ompy/vector.py [hunk 2]
```diff
@@ -424,6 +424,67 @@ def to_keV(self) -> Vector:
...
+
+            Args:
+                factor: A factor to multiply to the resulting vector.
+                        Possible values are None, a float or string 'de',
+                        meaning that the factor is determined by the
+                        bin width of the vector.
```
Comment (fzeiser) [inline ompy/vector.py:463]: Add how the factor is determined? "The factor preserves (...)", or just the formula?

File: ompy/vector.py [hunk 3]
```diff
@@ -424,6 +424,67 @@ def to_keV(self) -> Vector:
         else:
             raise NotImplementedError("Units must be keV or MeV")
 
+    def closest(self, E: ndarray, side: Optional[str]='right',
+                inplace=False) -> Union[Vector, None]:
+        """ Returns a vector with the values are set by
+            E[i] <= E[j] < E[i+1], values[i].
```
Comment (fzeiser) [inline ompy/vector.py:430]: sorry, but I don't understand this. I struggle especially with the `values[i]` part. It is not clear to me what the function does

**Comment by fzeiser** [comment]
Just another note, even though it's slightly off topic: I see that you often struggle with the coding style. For me, it helped a lot to get a prp8 linter. If you still use Sublime, I can recommend [SublimeLinter](https://github.com/SublimeLinter/SublimeLinter)

File: ompy/vector.py (hunk 3 above)
Comment (fzeiser) [reply ompy/vector.py:430]: Do I get it correctly that this returns a vector with the *new bins set to the (old) bin values closest in energy*?

Comment (fzeiser) [reply ompy/vector.py:430]: I wonder whether you could/should remove `E[i] <= E[j] < E[i+1]` here ,as you choose from the "side" which one to take?

**Comment by vetlewi** [comment]
> Just another note, even though it's slightly off topic: I see that you often struggle with the coding style. For me, it helped a lot to get a prp8 linter. If you still use Sublime, I can recommend [SublimeLinter](https://github.com/SublimeLinter/SublimeLinter)

Thanks!

File: ompy/vector.py (hunk 3 above)
Comment (vetlewi) [reply ompy/vector.py]: I've updated the docstring. It should be clearer what the function does.

File: ompy/vector.py (hunk 2 above)
Comment (vetlewi) [reply ompy/vector.py]: Fixed

File: ompy/vector.py [hunk 4]
```diff
@@ -424,6 +424,67 @@ def to_keV(self) -> Vector:
...
+        else:
+            return Vector(values=values, E=E, std=std, units=self.units)
+
+    def cumulative(self, factor: Optional[Union[float, str]]=None,
+                   inplace=False) -> Union[Vector, None]:
+        """ Calculate the cumulative sum of the vector.
+
+            Args:
+                factor: A factor to multiply to the resulting vector.
+                        Possible values are None, a float or string 'de',
```
[the remark this answers is not in the document]
Comment (vetlewi) [reply ompy/vector.py]: Fixed.

File: ompy/vector.py (hunk 1 above)
Comment (vetlewi) [reply ompy/vector.py]: Fixed.

**Pull request merged**